var/spack/repos/builtin/packages/py-tensorflow-probability/package.py

```diff
@@ -58,23 +58,23 @@ class PyTensorflowProbability(Package):
     depends_on("py-dm-tree", when="@0.12:", type=("build", "run"))
 
     # tensorflow_probability/python/__init__.py
-    # TODO: reactivate the JAX versions once the JAX package is available with newer versions
-    #  also add jaxlib as a dependency
-
-    variant("py-tensorflow", default=False, description="Build with TensorFlow support")
-    with when("+py-tensorflow"):
-        # TODO(0.24): requires tf-keras on Spack
-        # depends_on("py-tf-keras", when="@0.24:", type=("build", "run"))
-        # depends_on("py-tensorflow@2.16", when="@0.24", type=("build", "run"))
+
+    variant("tf", default=False, description="Build with TensorFlow support")
+    with when("+tf"):
+        depends_on("py-tf-keras@2.18:", when="@0.25:", type=("build", "run"))
+        depends_on("py-tf-keras@2.17:", when="@0.24:", type=("build", "run"))
+        depends_on("py-tensorflow@2.17", when="@0.24", type=("build", "run"))
         depends_on("py-tensorflow@2.15", when="@0.23", type=("build", "run"))
         depends_on("py-tensorflow@2.14:2", when="@0.22", type=("build", "run"))
         depends_on("py-tensorflow@2.13:2", when="@0.21", type=("build", "run"))
         depends_on("py-tensorflow@2.12:2", when="@0.20", type=("build", "run"))
         depends_on("py-tensorflow@2.11:2", when="@0.19", type=("build", "run"))
 
     # jaxlib is not required, as it's already a dependency of py-jax
```

Comment:
May be worth adding comment specifying where to look for supported JAX versions.

Reply:
Added a comment, also in tf-keras to make it clearer.

For JAX, there isn't a minimum version, just a "stable against". So the minimum is the stable, but this could be relaxed if needed. However, I think it's rather uncommon to _require_ a newer version of package Y with an older version of X (Y is still available for older X). So this should help resolve things and make sure it works.